lib/app/services/auth.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:convert';
+
+import 'package:dio/dio.dart';
+import 'package:overnites/app/services/bearer_token.dart';
+
+class AuthService {
+  AuthService(this._dio, this._bearerTokenService);
+
+  final Dio _dio;
+  final TokenService _bearerTokenService;
+
+  String decodeBase64(String str) {
+    //'-', '+' 62nd char of encoding,  '_', '/' 63rd char of encoding
+    var output = str.replaceAll('-', '+').replaceAll('_', '/');
+
+    switch (output.length % 4) {
+      // Pad with trailing '='
+      case 0: // No pad chars in this case
+        break;
+      case 2: // Two pad chars
+        output += '==';
+        break;
+      case 3: // One pad char
+        output += '=';
+        break;
+      default:
+        throw Exception('Illegal base64url string!"');
+    }
+
+    return utf8.decode(base64Url.decode(output));
+  }
+
+  Future<bool> verifyToken(String accessToken) async {
```

Comment:
Private and static?

Reply:
I will be moving this to the bearer token factory